src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -170,18 +170,20 @@ impl Window {
             )
             .map_err(|_| os_error!(OsError::WaylandMisc("failed to create window.")))?;
 
-        // Set CSD frame config
+        // Set CSD frame config from theme if specified
         #[cfg(feature = "sctk-adwaita")]
         {
-            let theme = platform_attributes.csd_theme.unwrap_or_else(|| {
-                let env = std::env::var(WAYLAND_CSD_THEME_ENV_VAR).unwrap_or_default();
-                match env.to_lowercase().as_str() {
-                    "dark" => Theme::Dark,
-                    _ => Theme::Light,
+            let theme = platform_attributes.csd_theme.or_else(|| {
+                match std::env::var(WAYLAND_CSD_THEME_ENV_VAR) {
+                    Ok(t) if t.eq_ignore_ascii_case("dark") => Some(Theme::Dark),
+                    Ok(t) if t.eq_ignore_ascii_case("light") => Some(Theme::Light),
+                    _ => None,
```

Comment:
I'm not sure why change it that way. You should add explicit `"light"` and return `None` from catch-all.

Reply:
> You should add explicit "light" and return None from catch-all.

Isn't that what is already written? The change allows the `None` path to use upstream defaults, instead of always setting.

I kept existing case insensitivity, just used `eq_ignore_ascii_case` instead of `to_lowercase` for efficiency.

Reply:
Then you can do something like

```rust
let mut env_var = std::env::var(WAYLAND_CSD_THEME_ENV_VAR).unwrap_or_default();
// Can modify only first 5 chars with `chars` and take, but it shouldn't matter.
env_var.make_ascii_lowercase();
match env_var {
  "dark" => Some(Theme::Dark),
  "light" => Some(Theme::Light),
  _ => None,
}
```

So you'd have 1 conversion and one allocation, unlike 2 conversions like you have now?

They are pretty much the same though. Maybe I just don't like how `Ok(t) if t.eq_ignore_ascii_case("dark")` looks.

Reply:
Your version just seems less direct to me and more code. `eq_ignore_ascii_case` avoids any byte conversion when the len doesn't match so I think the code is 2 comparisons max 1 of which can iterate over max 5 bytes (to compare to "light").

This kind of analysis is almost certainly not worth the effort, though can be interesting. Originally removing the previously existing 2nd allocation (`to_lowercase`) seemed worthwhile since, at the time, it seemed not much effort.

:shrug: Overall not a big deal, I don't understand the dislike of match guards, but feel free to edit it to whatever you like!

Reply:
Could go with that then. My point is to have something more readable for the code that is cold and never executed, so it's easier to grasp it.

```rust
let env_var = std::env::var(WAYLAND_CSD_THEME_ENV_VAR).unwrap_or_default();
if env_var.eq_ignore_ascii_case("dark") {
    Some(Theme::Dark)
} else if env_var.eq_ignore_ascii_case("light") {
    Some(Theme::Light)
} else {
    None
}
```

Reply:
Oh, besides that. Sorry for bothering you, but could you add a method like `try_into` for winit `Theme` here, so we can simply write.

```rust
// If not specified use upstream default (automatically selects theme)
if let Some(theme) = platform_attributes.csd_theme.try_into() {
    window.set_frame_config(theme.into());
}
```

Reply:
I think a `try_into` on `csd_theme` option that also checks an env-var may be a bit surprising. I've refactored to add an use a `impl TryFrom<&str> for Theme` though, wydt?